### ISSUE TITLE
Generated type definitions may have namespaces conflicts

### ISSUE THREAD

**Issue by bravik** [overblog/GraphQLBundle#863]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Version/Branch   | 0.13,0.14

I created 2 cusstom validation constraints, with following names:
```php
XXX\V2\Common\Infrastructure\Validator\DatesAreWithinInterval\Constraint
XXX\V2\Common\Infrastructure\Validator\AtLeastOneFieldPresent\Constraint;
```
I usually prefer organizing files by namepsace against naming files like `DatesAreWithinIntervalConstraint`.

And I'm using it for input validation in different files/types:
```yaml
SomeInput:
  type: input-object
  config:
    fields:
      #...
      general:
        type: 'Void'
        validation:
          - XXX\V2\Common\Infrastructure\Validator\AtLeastOneFieldPresent\Constraint: ~
```
I get: an error:
> Compile Error: Cannot use XXX\V2\Common\Infrastructure\Validator\DatesAreWithinInterval\Constraint as Constraint because the name is already in use

Here is one of the generated definition files:
![image](https://user-images.githubusercontent.com/1075618/122146205-f45d3780-ce99-11eb-8144-f7eeef545807.png)

1) Why unused constraints are imported?
2) If there is a namespace collision possible, why not generating a file with FQCN instead of `use` import?

**Comment by murtukov** [comment]
Hi @bravik, thanks for reporting the issue. Unfortunately I don't have much time now to investigate the bug, but could you check if it produces the same error, when you try to apply the same custom constraint but for `object` GraphQL type?

**Comment by bravik** [comment]
Tried to update to v0.14, but got same error with standard symfony validation constrints:
![image](https://user-images.githubusercontent.com/1075618/146530360-63eca468-7d97-4d03-be7a-e173a3c80cee.png)

**Comment by murtukov** [comment]
Hi @bravik. I will look into that issue in the next days. It should be fixed earlier, I just didn't have much time for it.

**Comment by bravik** [comment]
@murtukov I've made a fix for that:
https://github.com/overblog/GraphQLBundle/pull/968
check it out

**Comment by bravik** [comment]
Fixed in #968
